gpt_index/vector_stores/metal.py

```diff
@@ -0,0 +1,116 @@
+import math
+from typing import Any, Dict, List, Optional
+from gpt_index.data_structs.node_v2 import Node
+from gpt_index.vector_stores.types import VectorStore, NodeEmbeddingResult, VectorStoreQuery, VectorStoreQueryResult
+
+
+class MetalVectorStore(VectorStore):
+    def __init__(self, api_key: str, client_id: str, index_id: str, filters: Optional[Dict[str, Any]] = None):
+        """Init params."""
+        import_err_msg = (
+            "`metal_sdk` package not found, please run `pip install metal_sdk`"
+        )
+        try:
+            import metal_sdk  # noqa: F401
+        except ImportError:
+            raise ImportError(import_err_msg)
+        from metal_sdk.metal import Metal   # noqa: F401
+
+
+        self.api_key = api_key
+        self.client_id = client_id
+        self.index_id = index_id
+        self.filters = filters
+
+        self.metal_client = Metal(api_key, client_id, index_id)
+        self.stores_text = True
+        self.is_embedding_query = True
+
+    @classmethod
+    def from_dict(cls, config_dict: Dict[str, Any]) -> "MetalVectorStore":
```

Comment:
We recently removed `config_dict` and `from_dict` to make the vector store interface simpler. 
You can just remove them here.